aaelf32/aaelf32.rst

```diff
@@ -0,0 +1,3317 @@
+capabilities required by the executable file.  The segment is optional, but if
+present it must appear before segment of type PT_LOAD.  The platform
+independent parts of this segment are described in `Platform architecture compatibility data`_.
+
+PT_ARM_EXIDX (alias PT_ARM_UNWIND) describes the location of a program’s unwind
+tables.
+
+.. rubric:: p_flags
+
+There are no processor-specific flags.  All bits in the PT_MASKPROC part of
+this field are reserved to future revisions of this specification.
+
+Platform architecture compatibility data
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This data describes the platform capabilities required by an executable file.
+It can be constructed by a linker using the attributes [Addenda32_] found in
+its input relocatable files, or otherwise.
+
+If this segment is present it shall contain at least one 32-bit word with
+meaning defined by `Common architecture compatibility data masks`_,
+`Architecture compatibility data formats`_, `Architecture profile compatibility
+data`_, and `Architecture version compatibility data`_ below.
+
+.. _Common architecture compatibility data masks:
+
+.. class:: aaelf32-arch-compat
+
+.. table:: Common architecture compatibility data masks
+
+  +------------------------+------------+---------------------------------------------------------------------------------------------+
+  | Name                   | Value      | Meaning                                                                                     |
+  +========================+============+=============================================================================================+
+  | PT_ARM_ARCHEXT_FMTMSK  | 0xff000000 | Masks bits describing the format of data in subsequent words. The masked value is           |
+  |                        |            | described in `Architecture compatibility data formats`_, below.                             |
+  +------------------------+------------+---------------------------------------------------------------------------------------------+
+  | PT_ARM_ARCHEXT_PROFMSK | 0x00ff0000 | Masks bits describing the architecture profile required by the executable. The masked value |
+  |                        |            | is described in `Architecture profile compatibility data`_, below.                          |
+  +------------------------+------------+---------------------------------------------------------------------------------------------+
+  | PT_ARM_ARCHEXT_ARCHMSK | 0x000000ff | Masks bits describing the base architecture required by the executable. The masked value is |
+  |                        |            | described in `Architecture version compatibility data`_, below.                             |
+  +------------------------+------------+---------------------------------------------------------------------------------------------+
+
+The following table lists the architecture compatibility data formats defined
```

Comment:
For a table with a title (introduced with ".. table:: Some text here"), I haven't been able to find any way to avoid the large margin above the title. Setting table.spaceAbove to 0 in the rst2pdf style file has a tiny effect on that margin, but apparently most of it comes from somewhere else.

For this section, what I've done is to take the titles off the tables themselves, and move them to ..rubric lines before the tables, and then each paragraph saying "The following table" lives _between_ the rubric and the table itself, which should help.

Reply:
ah, yea, that's nice.